runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/provider/MuleExtensionModelDeclarer.java

```diff
@@ -578,6 +584,19 @@ private void declareParseTemplate(ExtensionDeclarer extensionDeclarer) {
         .withExpressionSupport(NOT_SUPPORTED)
         .describedAs("The mime type to be assigned to the result generated when parsing the template, e.g. text/plain or application/json");
 
+    if (isParseTemplateUseLegacyDefaultTargetValue()) {
+      parseTemplate.onParameterGroup(OUTPUT)
+          .withOptionalParameter(TARGET_VALUE_PARAMETER_NAME)
+          .ofType(STRING_TYPE)
+          .defaultingTo("#[message]")
+          .withExpressionSupport(REQUIRED)
+          .describedAs(TARGET_VALUE_PARAMETER_DESCRIPTION)
+          .withRole(BEHAVIOUR)
+          .withDisplayModel(DisplayModel.builder().displayName(TARGET_VALUE_PARAMETER_DISPLAY_NAME).build())
+          .withLayout(LayoutModel.builder().tabName(ADVANCED_TAB).build())
+          .withModelProperty(new TargetModelProperty());
```

Comment:
I'm not sure I get what this model property does. Nobody seems to be checking for it.

Reply:
I ended up removing this ModelProperty, I can't think of any way this would affect anything, this is a private property and it is not even serialized.